src/engraving/dom/edit.cpp

```diff
@@ -6471,19 +6474,43 @@ void Score::undoAddCR(ChordRest* cr, Measure* measure, const Fraction& tick)
                 // TODO solve this in read302.cpp.
                 tracks.push_back(strack);
             } else {
-                // linkedPart : linked staves within same part/instrument.
-                // linkedScore: linked staves over different scores via excerpts.
-                const bool linkedPart  = linked && (staff != ostaff) && (staff->score() == ostaff->score());
-                const bool linkedScore = linked && (staff != ostaff) && (staff->score() != ostaff->score());
-                for (track_idx_t track : muse::values(mapping, strack)) {
+                std::vector<track_idx_t> mappedTracks = muse::values(mapping, strack);
+                if (mappedTracks.empty()) {
+                    // This is a linked staff in a part and the element has been added to a linked staff in the main score
+                    track_idx_t track = staffIdx * VOICES + (strack % VOICES);
+                    track_idx_t mappedTrack = muse::value(mapping, track, muse::nidx);
+                    mappedTrack = (mappedTrack != muse::nidx) ? mappedTrack : track;
+                    if (staff->isVoiceVisible(strack % VOICES)) {
+                        mappedTracks.push_back(mappedTrack);
+                    }
+                }
+                for (track_idx_t track : mappedTracks) {
+                    // linkedPart : linked staves within same part/instrument.
+                    // linkedScore: linked staves over different scores via excerpts.
+                    const bool linkedPart  = linked && (staff != ostaff) && (staff->score() == ostaff->score());
+                    const bool linkedScore = linked && (staff != ostaff) && (staff->score() != ostaff->score());
```

Comment:
I know that this was there before, but while we're here, let's give these variables proper names such as isLinkedWithinSameScore and isLinkedToDifferentScore or something like that, rathert than use comments to clarify the meaning of poor names :)

Reply:
I've done what I should have in the first place and made these changes in addition to refactoring this into a method within `Staff`, as this is duplicated in `undoAddElement`.